src/components/Route/CreateRoute.tsx

```diff
@@ -0,0 +1,195 @@
+    return getAllLeadclimbRouteClassifiers();
+  return getAllTopropeRouteClassifiers();
+};
+
+const Ropes = [
+  '1', '2', '3', '4', '5', '6', '7', '8', '9'
+];
+
+type refreshRoutesCallback = () => void;
+
+type CreateRouteProps = {
+  refreshRoutes: refreshRoutesCallback;
+  isOpen: boolean;
+  setIsOpen: (arg0: boolean) => void;
+};
+
+// TODO: implement User search component
+const CreateRoute = ({refreshRoutes, isOpen, setIsOpen}: CreateRouteProps) => {
+  const [name, setName] = useState<string>('');
+  const [type, setType] = useState<RouteType>();
+  const [rope, setRope] = useState<number>();
+  const [rawgrade, setRawgrade] = useState<number>();
+  const [description, setDescription] = useState<string>();
+  const [rules, setRules] = useState<RouteTech>();
+  const [routeColor, setRouteColor] = useState<RouteColor>();
+  const [confirmationBoxOpen, setConfirmationBoxOpen] = useState(false);
+  const [thumbnailFile, setThumbnailFile] = useState<File>();
+  const [setter, setSetter] = useState<User>();
+  const [overrideSetterBool, setOverrideSetterBool] = useState<boolean>(true);
+  const [overrideSetterString, setOverrideSetterString] = useState<string>('');
+  const [formError, setFormError] = useState(false);
+
+  function handleFileSelect(event: React.ChangeEvent<HTMLInputElement>) {
+    if (event.target.files === null)
+      return;
+    setThumbnailFile(event.target.files[0]);
+  }
+
+  function checkRouteData(): boolean {
+    if (rawgrade === undefined || name.length === 0 || type === undefined 
+      || rope === undefined || rules === undefined || routeColor === undefined)
+    {
+      setFormError(true);
+      return false;
+    }
+    setFormError(false);
+    return true;
+  }
+
+  async function submitRoute() {
+    // pacify data
+    if (checkRouteData() === false)
+      return;
+    // check that no active route has the same name
+    createRoute(
+      {
+        name: name, 
+        classifier: new RouteClassifier(rawgrade!, type!), 
+        setter: setter,
+        setterRawName: overrideSetterString,
+        description: description,
+        rope: rope,
+        tech: rules,
+        thumbnail: thumbnailFile,
+        color: routeColor,
+      })
+      .then( (route) => {
+        // handle successful create route (refresh Routes page or pass route to parent component)
+        console.log('route created: ', route.docRef?.id);
+        refreshRoutes();
+        setIsOpen(false);
+      });
+  }
+
+  return (
+    <Popup open={isOpen} onClose={() => setIsOpen(false)} closeOnDocumentClick={false} nested modal>
+      <div>
+        <Popup nested modal open={confirmationBoxOpen} onClose={() => setConfirmationBoxOpen(false)}>
+          <VStack alignItems='center'>
+            <Text variant='header'>
+              Are you sure?
+            </Text>
+            <Text variant='body'>
+              Are you sure you want to create this route? The name and grade must be unique and cannot be changed later.
```

Comment:
so this is pretty good, are we planning to make the routes editable besides name/grade? I could back that